src/application-interface/src/ApplicationInterface.cpp

```diff
@@ -39,8 +39,9 @@ void ApplicationInterface::setSystemConnectionState(ConnectionState state) {
 }
 
 void ApplicationInterface::setSystemDeviceState(DeviceState state) {
-    LockGuard lock(m_mutex);
+    m_mutex.lock();
```

Comment:
Out of curiosity, why was this changed? Is it an issue with the LockGuard from Propolis or did the member really needed to be accessed earlier than the return of the function?

Reply:
Or before the syscall to setHexDisplay I would guess?

Reply:
Yea it's because setHexDisplay calls it so there would be a deadlock